pkg/moq/template.go

```diff
@@ -17,16 +17,6 @@ import (
 )
 
 {{ range $i, $obj := .Objects -}}
-var (
-{{- range .Methods }}
-	lock{{$obj.InterfaceName}}Mock{{.Name}}	sync.RWMutex
-{{- end }}
-)
-
-// Ensure, that {{.InterfaceName}}Mock does implement {{.InterfaceName}}.
-// If this is not the case, regenerate this file with moq.
-var _ {{$sourcePackagePrefix}}{{.InterfaceName}} = &{{.InterfaceName}}Mock{}
```

Comment:
I'm studying some golang code and get here in this PR =)
I'm trying to figure out why you remove this validation. It isn't important anymore?

Reply:
Not removing just moving it to inside the struct instead of being a global state variable. It has the same value but more contained. And yes, still important. This don't fix any kind of issue, the code works as it is, but it's better to keep the mutex inside the only place it's used.

Reply:
@diegobernardes That is not true. Lines 26 - 28 are removed completely by your PR. With this change, the compiler does no longer check upon compilation, if the mock actually implements the interface. This is especially useful if you work with an IDE, that compiles the code on save and highlights errors.

Reply:
Yes those lines remove, and the lines 53-55 add. The suggestion was just to remove the global state. What's the benefit of having a global state lock if the only use is inside the struct?

Reply:
@diegobernardes I do not agree with you. Lines 53 - 55 are the replacement for lines 20 - 24.

There is no replacement for the lines 26 - 28 in your commit. I added these lines in #59 (see  https://github.com/matryer/moq/pull/59/files#diff-a999f10d75067f9ce86eea4b883524e9R26-R29). Please checkout my [comment](https://github.com/matryer/moq/pull/59#issuecomment-397174241) for a more detailed reasoning.